asf_public_discourse_home_decarbonisation/analysis/buildhub/extract_faqs.py

```diff
@@ -0,0 +1,245 @@
+"""
+This script is designed to extract frequently asked questions (FAQs) and specific patterns from a given forum text dataset.
+Key Functionalities:
+    - Extract Questions: Identifies and extracts questions from text based on criteria such as the number of words and minimum sentence length.
+    - Extract Specific Patterns: Targets and extracts specific patterns or phrases, tailored by the `extract_idk` function.
+    - Configurable: Allows runtime configuration through command-line arguments for flexible usage in different scenarios.
+To run this script:
+    'python extract_faqs.py --forum <forum> --category <category> --collection_date_time <collection_date_time>'
+where:
+    <forum>: The forum that we are looking at (e.g. bh or MSE).
+    <category>: The category within the forum (e.g. energy).
```

Comment:
This won't work for BH, as it has a different name for all data. I'm adding both in but obviously will want to standardise this at some point (along with the datetimes)

Reply:
Just so you know, in my PR, I created a new getter that puts together data from all BH sub-forums. It was handy for my topic analysis and it might be handy for future work.